tuist/tuist
# Transitive SDK Dependencies

Resolves https://github.com/tuist/tuist/issues/438

### Short description 📝

SDK dependencies for static frameworks or libraries were not previously being linked transitively.

### Solution 📦

Traverse the target node graph for the SDK dependencies in a similar way that we do already for static libraries and frameworks.

### Implementation 👩‍💻👨‍💻

- [X ] Update `TuistGenerator` to traverse graph recursively for SDK dependencies 
- [X ] Update `Graph+TestData` to support SDKs
- [X ] Add unit tests to check that transitive dependencies are added without duplicates for static libraries
- [X ] Update `ios_app_with_sdk` fixture to include transitive SDK dependency cases similar to the one failing in the issue raised
- [X ] Add acceptance test

### Test Plan

- Run `tuist generate` within `fixtures/ios_app_with_sdk`
- Verify that StaticFrameworkTests within the StaticFramework project has `libc++.tbd`
- Verify that StaticFrameworkTests builds and the tests pass
